### PR TITLE
OIDC default scopes

### PR DESCRIPTION
**Pull request by dirkkul**
- Scopes can now be given as string or list
- Adds support of default scopes from weaviate that are added with the AUTHENTICATION_OIDC_SCOPES env variable (not documented yet)

### CODE REVIEW

File: weaviate/connect/authentication.py [hunk 1]
```diff
@@ -94,14 +100,24 @@ def _get_session_user_pw(self, config: AuthClientPassword) -> OAuth2Session:
...
 
     def _get_session_client_credential(self, config: AuthClientCredentials) -> OAuth2Session:
+        scope: List[str] = self._default_scopes.copy()
+
+        # remove openid scopes from the scopes returned by weaviate (these are returned by default). These are not
+        # accepted by some providers for client credentials
+        if "openid" in scope:
+            scope.remove("openid")
+        if "email" in scope:
+            scope.remove("email")
```
Comment (parkerduckworth) [inline weaviate/connect/authentication.py:110]: You could just filter out the unwanted scopes, that way the list is only traversed once:
```
scope = list(filter(lambda s: s != "openid" and s != "email", scope))
```

File: weaviate/connect/authentication.py [hunk 2]
```diff
@@ -21,13 +21,17 @@
 
 class _Auth:
     def __init__(
-        self, response: Dict[str, str], auth_config: AuthCredentials, connection: BaseConnection
+        self,
+        response: Dict[str, Union[str, List[str]]],
+        auth_config: AuthCredentials,
```
[the remark this answers is not in the document]
Comment (dirkkul) [reply weaviate/connect/authentication.py]: good idea, done

File: weaviate/connect/authentication.py (hunk 1 above)
Comment (dirkkul) [reply weaviate/connect/authentication.py]: done!

File: weaviate/auth.py [hunk 3]
```diff
@@ -1,22 +1,31 @@
...
 
 @dataclass
 class AuthClientCredentials:
     """Authenticate for the Client Credential flow using client secrets.
 
     Acquire the client secret from your identify provider and set the appropriate scope. The client includes hardcoded
     scopes for Azure, otherwise it needs to be supplied.
+    Scopes can be given as:
+      - List of strings: ["scope1", "scope2"]
+      - space separated string: "scope1 scope2"
```
[the remark this answers is not in the document]
Comment (dirkkul) [reply weaviate/auth.py:20]: I think a list is the better way to signal to the user how to provide multiple scopes, but I didn't want to break any existing setups

[the remark this answers is not in the document]
Comment (parkerduckworth) [reply weaviate/auth.py:20]: Got it, I didn't realize there was already a release which supports a string `scope` field

**Review by parkerduckworth — APPROVED** [review]

**Pull request merged**